ui/src/core_components/embeds/CoreGoogleMaps.vue

```diff
@@ -0,0 +1,223 @@
+				default: '',
+				desc: "ID of map from Google Cloud Console, needed for markers",
+				type: FieldType.Text,
+			},
+			mapType: {
+				name: "Map type",
+				default: 'roadmap',
+				type: FieldType.Text,
+				desc: "One of 'roadmap', 'satellite', 'hybrid' or 'terrain'",
+			},
+			zoom: {
+				name: "Zoom",
+				default: "8",
+				type: FieldType.Number,
+			},
+			lat: {
+				name: "Latitude",
+				default: "37.79322359164316",
+				type: FieldType.Number,
+			},
+			lng: {
+				name: "Longitude",
+				default: "-122.39999318828129",
+				type: FieldType.Number,
+			},
+			markers: {
+				name: "Markers",
+				default: JSON.stringify(markersDefaultData),
+				desc: "Markers data",
+				type: FieldType.Object,
+			},
+			cssClasses,
+		},
+		events: {
+			"gmap-marker-click": {
+				desc: "Capture single clicks on markers.",
+			},
+			"gmap-click": {
+				desc: "Capture single click on map.",
+			},
+		},
+	},
+};
+</script>
+
+<script setup lang="ts">
+import { Ref, inject, ref, onMounted, watch, computed } from "vue";
+import injectionKeys from "../../injectionKeys";
+import { Loader } from "@googlemaps/js-api-loader"
+
+const rootEl: Ref<HTMLElement> = ref(null);
+const mapEl: Ref<HTMLElement> = ref(null);
+const fields = inject(injectionKeys.evaluatedFields);
+let map : google.maps.Map | null = null;
+let markers = [];
+
+const center = computed<{ lat: number; lng: number; }>(() => ({ lat: fields.lat.value, lng: fields.lng.value }));;
+const mapType = computed(() => (
+	['roadmap', 'satellite', 'hybrid', 'terrain'].includes(fields.mapType.value) 
+		? fields.mapType.value 
+		: 'roadmap'
+));
+
+
+const initMap = async () => {
+	clearMarkers();
+	const loader = new Loader({
+		apiKey: fields.apiKey.value,
+		version: "weekly",
+	});
+	await loader.load();
+	map = new google.maps.Map(mapEl.value, {
+		center: center.value,
+		mapTypeId: mapType.value,
+		zoom: 8,
+		mapId: fields.mapId.value,
+	});
+
+	map.addListener('click', (e: any) => {
+		const event = new CustomEvent("gmap-click", {
+			detail: {
+				payload: e.latLng.toJSON(),
+			},
+		});
+		rootEl.value.dispatchEvent(event);
+	});
+
+	buildMarkers();
+}
+
+const buildMarkers = async () => {
+	if (!fields.mapId.value) return;
+	if (!fields.markers.value) return;
+	const { AdvancedMarkerElement } = await google.maps.importLibrary("marker") as google.maps.MarkerLibrary;
+	clearMarkers();
+	try {
+		const markersData = fields.markers.value as ({ lat: number; lng: number; name: string; }[])
+		markersData.forEach((markerData) => {
+			const marker = new AdvancedMarkerElement({
+				position: {
+					lat: markerData.lat,
+					lng: markerData.lng,
+				},
+				map,
+				title: markerData.name,
+			});
+
+			markers.push(marker);
+
+			marker.addListener('click', () => {
+				const event = new CustomEvent("gmap-marker-click", {
+					detail: {
+						payload: markerData,
+					},
+				});
+				rootEl.value.dispatchEvent(event);
+			});
+		});
+	} catch (e) {
+		console.log(e);
```

Comment:
Since it's an error let's switch to `console.error`.

Reply:
reason for this catch not exist anymore - removed